tests/moduletool/test_install.py

```diff
@@ -577,8 +565,6 @@ def test_project_install_modules_cache_invalid(
     message: str = (
         f"Compiler has loaded module {dependency_module}=={v2_version}.dev0 but {dependency_module}=={v2_version} has"
         " later been installed as a side effect."
-        if preinstall_v2
-        else f"Compiler has loaded module {dependency_module} as v1 but it has later been installed as v2 as a side effect."
```

Comment:
This error is no longer produced, because all required import statements were added. The state of the module cache is restored via this condition: https://github.com/inmanta/inmanta-core/blob/00a0b81d7c1b2d9c8b1bf8f4e9c85ab930f4dbde/src/inmanta/module.py#L1646

Reply:
I'm not convinced this is no longer reachable. I just think the test approach is too naive for the improved project loading. If the v2 module somehow gets installed after loading this should still trigger. I'm going to have a look to see if I can reproduce this.

Reply:
Ok, I'm satisfied. I believe it's still reachable but only in the extremely unlikely scenario with the following dependencies:
`inmanta_v1_module -> some_python_package -> inmanta_v2_module`. I did discover a typo throughout this code (the existing code, not your changes) but I'll open a separate pull request for that.